src/PhpSpec/Listener/MethodNotFoundListener.php

```diff
@@ -27,12 +29,33 @@ class MethodNotFoundListener implements EventSubscriberInterface
     private $resources;
     private $generator;
     private $methods = array();
+    private $wrongMethodNames = array();
+    /**
+     * @var NameCheckerInterface
+     */
+    private $nameChecker;
 
-    public function __construct(IO $io, ResourceManagerInterface $resources, GeneratorManager $generator)
-    {
+    /**
+     * @param IO $io
+     * @param ResourceManagerInterface $resources
+     * @param GeneratorManager $generator
+     * @param NameCheckerInterface $nameChecker
+     */
+    public function __construct(
+        IO $io,
+        ResourceManagerInterface $resources,
+        GeneratorManager $generator,
+        NameCheckerInterface $nameChecker = null
+    ) {
         $this->io        = $io;
         $this->resources = $resources;
         $this->generator = $generator;
+
+        if (null === $nameChecker) {
```

Comment:
In other places we've done this with a ternary:

```
$this->nameChecker = $nameChecker ? $nameChecher : new ReservedWordsMethodNameChecker();
```

(one day we can use the `?:` operator when we drop old PHP versions